spring-cloud-gcp-autoconfigure/src/test/java/org/springframework/cloud/gcp/autoconfigure/pubsub/it/PubSubChannelAdaptersIntegrationTests.java

```diff
@@ -150,17 +172,50 @@ public void sendAndReceiveMessageManualAck() {
 
 				Message<?> message = channel.receive(10000);
 				assertThat(message).isNotNull();
-				AckReplyConsumer acker =
-						(AckReplyConsumer) message.getHeaders().get(GcpPubSubHeaders.ACKNOWLEDGEMENT);
-				assertThat(acker).isNotNull();
-				acker.nack();
+				BasicAcknowledgeablePubsubMessage origMessage =
+						(BasicAcknowledgeablePubsubMessage) message.getHeaders().get(GcpPubSubHeaders.ORIGINAL_MESSAGE);
+				assertThat(origMessage).isNotNull();
+				origMessage.nack();
+
+				message = channel.receive(10000);
+				assertThat(message).isNotNull();
+				origMessage = (BasicAcknowledgeablePubsubMessage)
+						message.getHeaders().get(GcpPubSubHeaders.ORIGINAL_MESSAGE);
+				assertThat(origMessage).isNotNull();
+				origMessage.ack();
+
 				message = channel.receive(10000);
+				assertThat(message).isNull();
+			}
+			finally {
+				PubSubAdmin pubSubAdmin = context.getBean(PubSubAdmin.class);
+				pubSubAdmin.deleteSubscription((String) context.getBean("subscriptionName"));
+				pubSubAdmin.deleteTopic((String) context.getBean("topicName"));
+			}
+		});
+	}
+
+	@Test
+	public void sendAndReceiveMessageManualAckThroughAcknowledgementHeader() {
+		this.contextRunner.run(context -> {
+			try {
+				context.getBean(PubSubInboundChannelAdapter.class).setAckMode(AckMode.MANUAL);
+				context.getBean("inputChannel", MessageChannel.class).send(
+						MessageBuilder.withPayload("I am a message.".getBytes()).build());
+
+				PollableChannel channel = context.getBean("outputChannel", PollableChannel.class);
+
+				Message<?> message = channel.receive(10000);
 				assertThat(message).isNotNull();
-				acker = (AckReplyConsumer) message.getHeaders().get(GcpPubSubHeaders.ACKNOWLEDGEMENT);
+				AckReplyConsumer acker =
+						(AckReplyConsumer) message.getHeaders().get(GcpPubSubHeaders.ACKNOWLEDGEMENT);
```

Comment:
I think the `sendAndReceiveMessageManualAckThroughAcknowledgementHeader()` method has to be marked with the `@SuppressWarnings("deprecation")` since `GcpPubSubHeaders.ACKNOWLEDGEMENT` is `@Deprecated`.

Reply:
Yep. I need to re-learn to pay attention to IDE warnings; I've been spoiled by the internal tooling style checks that show up right in the review tool.